### PR TITLE
adding kubernetes networking notes from #360

### PR DESCRIPTION
**Pull request by rigzba21**
Hi all! Excited to make my first PR to Dask-Gateway!

As mentioned in  https://github.com/dask/dask-gateway/issues/435 adding networking notes identified in https://github.com/dask/dask-gateway/issues/360#issuecomment-741767913

Thanks!

### CODE REVIEW

**Comment by rigzba21** [comment]
It looks like these github workflows may be experiencing https://github.com/dask/dask-gateway/issues/422.

File: docs/source/install-kube.rst [hunk 1]
```diff
@@ -24,6 +24,21 @@ When running on Kubernetes, Dask Gateway is composed of the following components
...
+- ``api``: The **Gateway API Server**
+- ``scheduler`` & ``worker``: User's Dask Scheduler and Worker
+- ``controller``: The Kubernetes **Gateway Controller** for managing Dask-Gateway resources
+
+Network communications happen in the following manner:
+- The ``traefik`` pods proxy connections to the ``api`` pods on port 8000, and ``scheduler`` pods on ports 8786 and 8787.
+- The ``api`` pods send api requests to the ``scheduler`` pods over port 8788.
+- If using JupyterHub Auth., the ``api`` pod sends requests to the JupyterHub server to authenticate.
+- Depending on the configuration, requests go directly to the JupyterHub pods through service lookup or through the JupyterHub Proxy.
+- The ``worker`` pods communicate between themselves and the ``scheduler`` pod over ports 8786 and 8787.
```
[the remark this answers is not in the document]
Comment (rigzba21) [reply docs/source/install-kube.rst]: @jacobtomlinson updated per your recommendations!

**Comment by rigzba21** [comment]
Hi all, I just wanted to see if there was a status on this getting merged? Thanks!

**Comment by martindurant** [comment]
Yep, all good - sorry this got forgotten.

**Pull request merged**